nixos/modules/security/apparmor.nix

```diff
@@ -42,22 +51,55 @@ in
         description = ''
           AppArmor policies.
         '';
-        type = types.attrsOf (types.submodule ({ name, config, ... }: {
-          options = {
-            enable = mkDisableOption "loading of the profile into the kernel";
-            enforce = mkDisableOption "enforcing of the policy or only complain in the logs";
-            profile = lib.mkOption {
-              description = "The policy of the profile.";
-              type = types.lines;
-              apply = pkgs.writeText name;
-            };
-          };
-        }));
-        default = {};
+        type = types.attrsOf (
+          types.submodule (
+            {
+              name,
+              config,
+              options,
+              ...
+            }:
+            {
+              options = {
+                state = lib.mkOption {
+                  description = "The state of the profile as applied to the system by nix";
+                  type = types.enum [
+                    "disable"
+                    "complain"
+                    "enforce"
+                  ];
+                  # should enforce really be the default?
+                  # the docs state that this should only be used once one is REALLY sure nothing's gonna break
+                  default = "enforce";
```

Comment:
alternatively if you don't set a default users will have to specify a value explicitly for each policy

Reply:
I put `default = "enforce";` because this was default before this PR.
The problem is, setting a different default would mean behavior would *silently* change in the background, i do not believe this is a good idea.
Not setting a default would mean existing configs relying on an old default would at least *fail loud*. However, it stands to reason this would actively break even more configs that would otherwise just silently use the new implementation with the old effective behavior.

Reply:
makes sense